helpers/iframe.js

```diff
@@ -0,0 +1,52 @@
+'use strict';
+
+const config = require('config');
+const uiUrl = config.get('ecosystem.ui');
+const apiUrl = config.get('ecosystem.api');
+const apiVersion = 'v4';
+
+const iframeScript = `
+    function replaceHref(apiHref='${apiUrl}') {
+        const currentIframeHref = new URL(document.location.href);
+        const urlOrigin = currentIframeHref.origin;
+        const urlFilePath = decodeURIComponent(currentIframeHref.pathname);
+
+        let urlFileDir = urlFilePath.split('/');
+        urlFileDir = urlFileDir.slice(2, urlFileDir.length-1).join('/');
+
+        const apiUrl = new URL(apiHref);
+        const apiUrlOrigin = apiUrl.origin;
+        const apiVersion = apiUrl.pathname.split('/')[1] || '${apiVersion}';
+
+        const anchors = document.getElementsByTagName('a');
+        for (let anchor of anchors) {
+          let originalHref = anchor.attributes.href.value;
+          const newUrl = apiUrlOrigin + '/' + apiVersion + '/' + urlFileDir + '/' + originalHref;
```

Comment:
encodeURI?

Reply:
This function does string concatenation to mask the urls to be prefix-ed from api.